pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -465,12 +465,36 @@ func (l *LibvirtDomainManager) initializeMigrationMetadata(vmi *v1.VirtualMachin
 }
 
 func (l *LibvirtDomainManager) cancelMigration(vmi *v1.VirtualMachineInstance) error {
-	if vmi.Status.MigrationState == nil || vmi.Status.MigrationState.Completed ||
-		vmi.Status.MigrationState.Failed || vmi.Status.MigrationState.StartTimestamp == nil {
 
+	domain, err := func() (*api.DomainSpec, error) {
+		l.domainModifyLock.Lock()
+		defer l.domainModifyLock.Unlock()
+
+		domName := api.VMINamespaceKeyFunc(vmi)
+		dom, err := l.virConn.LookupDomainByName(domName)
+		if err != nil {
+			return nil, err
+		}
+		defer dom.Free()
+
+		domain, err := l.getDomainSpec(dom)
+		if err != nil {
+			return nil, err
+		}
+		return domain, nil
+	}()
+
+	if err != nil {
+		return err
+	}
+
+	migration := domain.Metadata.KubeVirt.Migration
+	if migration == nil || migration.Completed ||
```

Comment:
@xpivarc could you explain how thi can solve the issue? One of the things which we see is that we have already this scenario on the VMI:

```
completed: true
failed: true
abortStatus: AbortInProgress
```

The change here looks to me like it would only ensure that the AbortInProgress status gets set, which seems to have happened on the VMI and the domain metadata.

Reply:
This is what is happening (At least for the job linked in the description. I think there are actually more possible paths...):
1. We call `CancelMigration` -> `AbortInProgress` get set and in async we try to abort it.
2. Monitor sees the migration was aborted. It sets `completed, failed, AbortSuccesful` and exit. (This can be seen in generated XML)
3. We call `CancelMigration` again(delay in domain propagation - hopefuly) -> The information is not yet reflected on vmi and therefore we set the `AbortInProgress` status again and try to abort it again. 

Now there is no monitor and nobody seems to get us from this state.

The change is not perfect. Here we only ensure we use more up-to-date information for the decision. It doesn't protect us from failure detected by monitor between getting the domain and starting the abort.

Reply:
I am tempted to fix this by moving the responsibilities. I think it would be better if the monitor is responsible for aborting. It would just need another channel for this. WDYT? Then we don't need to worry about race conditions and other issues around metadata...

Reply:
Sounds reasonable. The job knows if it got aborted or succeeded. :+1: